### DISCUSSION TITLE
Turning web server on  using a push button

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#11717] [Q&A] [answered]
On my device I have a push button defined as Button 1. I can reset my device by pressing this button for 40 seconds. It would be great if more functionality can be added to this button, for example: if a command is issued to turn off the web server from console,  and device is not configured to use MQTT, then there is no way to turn web server back on other than resetting the device, now if we could somehow program Button 1 to turn the web server back on, by let’s say when pressed twice, it would save the hassle of having to reset and then reconfigure the device afresh. Is it possible to do this? Can this feature be included by default in the bin file?

**Comment by barbudor** [comment, marked as answer]
You can use multipress https://tasmota.github.io/docs/Buttons-and-Switches/#multi-press-functions
or HOLD https://tasmota.github.io/docs/Buttons-and-Switches/#long-press

**Reply by NsinghP** [reply]
I see that all of it boils down to mastery of Rules and autoexec.bat file. So time to put my reading glasses on and hunker down until I understand how to do that!
